tapir/configuration/models.py

```diff
@@ -0,0 +1,69 @@
+from enum import Enum
+
+from django.db import models
+
+
+class ParameterTag(models.Model):
+    tag = models.CharField(max_length=64, primary_key=True)
+
+    def __str__(self):
+        return self.tag
+
+
+class ParameterDatatype(Enum):
+    INTEGER = "integer"
+    DECIMAL = "decimal"
+    BOOLEAN = "boolean"
+    STRING = "string"
+
+
+class Parameter(models.Model):
+    key = models.CharField(max_length=256, primary_key=True, editable=False)
+    description = models.CharField(max_length=512)
+    tags = models.ManyToManyField(ParameterTag)
+    datatype = models.CharField(
+        choices=map(lambda d: (d, d), ParameterDatatype), max_length=8, editable=False
+    )
+    value = models.CharField(max_length=4096, null=True)
+
+    def get_value(self):
+        if self.datatype == ParameterDatatype.INTEGER.value:
+            return int(self.value)
+        elif self.datatype == ParameterDatatype.DECIMAL.value:
+            return float(self.value)
+        elif self.datatype == ParameterDatatype.BOOLEAN.value:
+            return bool(self.value)
+        elif self.datatype == ParameterDatatype.STRING.value:
+            return self.value
+        else:
+            Exception("""Unknown parameter type: {}""".format(self.datatype))
+
+
+def get_config(key: Enum):
+    param = Parameter.objects.get(pk=key.value)
+    return param.get_value()
+
+
+def parameter_definition(
+    key: Enum,
+    description: str,
+    tags: list,
+    datatype: ParameterDatatype,
+    initial_value: str,
+):
+    converted_tags = list(map(lambda t: ParameterTag(t), tags))
+    for tag in converted_tags:
+        tag.save()
```

Comment:
I like to create objects through `ParameterTag.objects.create()` rather than create an instance and save, I think it makes it clearer when the DB gets written. So I'd do something like :
```python
tags = []
for tag in tags:
    tag = ParameterTag.objects.filter(tag=tag).first()
    tags.append(tag if tag else ParameterTag.objects.create(tag=tag))
```

Reply:
I forgot to annotate the function as transactional (so the DB gets written after successful execution of the function, not really a benefit then). 

What's the problem with using save() for createOrUpdate operations?

On second look I would optimize the function by selecting all needed tags (select * from parameterTag where tag in (tags)")